scripts/kustomize.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# kustomize.sh: a wrapper for the kubernetes binary used to make it easier to deploy
+#
+# the script will:
+# - download the appropriate binary from github
+# - verify that the checksum is ok
+# - cache the binary for further executions
+
+set -euo pipefail
+
+RELEASE=1.0.7
+
+# See here for the checksums below:
+# https://github.com/kubernetes-sigs/kustomize/releases/download/v1.0.6/checksums.txt
+CHECKSUM_LINUX=afac1af5a6a7c688d5665f729d4d3a48492836f5baf20ceb1e7b71211a626ca5
+CHECKSUM_DARWIN=28d0265d4139f65dbd2c01fb91da63fc13e7bba4fd3e19f41595a02a453da2c4
+CHECKSUM_WINDOWS=2e9b1b0600c906760a0fe04136185fb44356fab32605a7524978c0e6345fea1b
+CACHE_DIR=$HOME/.cache/kustomize
+RELEASE_DOWNLOAD_BASE=https://github.com/kubernetes-sigs/kustomize/releases/download/v$RELEASE
+
+# Detect my architecture
+UNAME=$(uname -ms)
+ARCH=""
+case "$UNAME" in
+	"Darwin x86_64")
+		ARCH="darwin_amd64"
+		CHECKSUM_EXPECTED="$CHECKSUM_DARWIN"
+		sha256() {
+			shasum -a 256 $1 | awk '{print $1}'
+		}
+		;;
+	"Linux x86_64")
+		ARCH="linux_amd64"
+		CHECKSUM_EXPECTED="$CHECKSUM_LINUX"
+		sha256() {
+			sha256sum $1 | awk '{print $1}'
+		}
+		;;
+	MINGW*)
```

Comment:
What is MINGW?

Reply:
MINGW is a port of bash and other Unix/Linux tools to Windows. It's what "Git BASH" for Windows uses.

The script being a bash script and the fact that Windows doesn't ship with bash makes this for Windows a bit special. I have the feeling that it can still be useful however. Many of my colleagues use Git BASH.